xla/pjrt/mlir_to_hlo.cc

```diff
@@ -223,6 +226,10 @@ absl::StatusOr<std::string> SerializeUsingVersionedStablehlo(
   // Legalize CHLO -> [StableHLO+Shape] -> StableHLO
   // Preserve higher-level ops with XLA support. To be replaced by composites.
   mlir::PassManager pm(context);
+  // Expand stablehlo complex math functions such as log_plus_one, etc.
+  pm.addNestedPass<mlir::func::FuncOp>(
```

Comment:
Do we need this one here as well? Not expanding during serialize will keep serialized artifacts smaller. I think all paths should go through the other function so we likely don't need this.

Reply:
Yes, apparently we do need it for CUDA path: removing the pass from SerializeUsingVersionedStablehlo will lead to previous behavior. FWIW, here's the output of compile result:
```python
>>> print(jnp.log1p.lower(jnp.float32(0)).compile().as_text())
HloModule jit_log1p, is_scheduled=true, entry_computation_layout={(f32[])->f32[]}, allow_spmd_sharding_propagation_to_parameters={true}, allow_spmd_sharding_propagation_to_output={true}, frontend_attributes={fingerprint_before_lhs="29db0bac90aebef6db93a00db5b57dfb"}

%wrapped_log-plus-one_computation (param_0: f32[]) -> f32[] {
  %param_0 = f32[] parameter(0)
  ROOT %log-plus-one.3.1 = f32[] log-plus-one(f32[] %param_0), metadata={op_name="jit(log1p)/jit(main)/log1p" source_file="<stdin>" source_line=1}
}

ENTRY %main.3 (Arg_0.1.0: f32[]) -> f32[] {
  %Arg_0.1.0 = f32[] parameter(0), metadata={op_name="x"}
  ROOT %wrapped_log-plus-one = f32[] fusion(f32[] %Arg_0.1.0), kind=kLoop, calls=%wrapped_log-plus-one_computation, metadata={op_name="jit(log1p)/jit(main)/log1p" source_file="<stdin>" source_line=1}
}

```

Reply:
The difference of CPU and CUDA paths is as follows:
```
CPU:
xla::ifrt::PjRtLoadedExecutable::Create 
  -> xla::TfrtCpuClient::Compile 
  -> xla::MlirToXlaComputation

CUDA:
xla::ifrt::PjRtLoadedExecutable::Create 
  -> xla::PjRtCApiClient::Compile 
  -> xla::Serialize 
  -> xla::SerializeUsingVersionedStablehlo
```
that is, `MlirToXlaComputation` is never called when on CUDA path.

Reply:
Can you add a todo comment for fixing this? This isn't expected. There should be a single path to XLA HLO and ideally should be the other method

Reply:
@GleasonK, re the todo comment, see issue https://github.com/openxla/xla/issues/20903